docs/source/models/spatial-knn.rst

```diff
@@ -0,0 +1,288 @@
+============================
+Spatial K Nearest Neighbours
+============================
+
+Intro
+###################
+
+Nearest neighbour problem is defined as finding an observation from a set S that is the most similar to the given
+observation O. The similarity is defined by a distance function d(O, S[i]). The most common distance function is the Euclidean.
+The nearest neighbour problem is a special case of the k-nearest neighbour problem, where k=1. The k-nearest neighbour
+problem is defined as finding k observations from a set S that are the most similar to the given observation O.
+
+.. figure:: ../images/nearest_neighbour.png
+   :figclass: doc-figure
+
+   Fig 1. Nearest neighbour problem
+
+We define spatial k-nearest neighbour problem as finding k observations from a set S that are the most similar to the
```

Comment:
This part is repeated from above:
"The k-nearest neighbour problem is defined as finding k observations from a set S that are the most similar to the given observation O."